### PR TITLE
Created tip-control component

### PR DESCRIPTION
**Pull request by stanislav-slavov**

### CODE REVIEW

File: src/components/tipRecords/TipRecord.vue [hunk 1]
```diff
@@ -34,8 +34,8 @@
         class="tip__article"
       >
         <a
-          :href="tip.url"
           target="_blank"
+          @click="goToUrl(tip.url)"
```
Comment (venimus) [inline src/components/tipRecords/TipRecord.vue:38]: why is that needed?
This way you can't see the link tooltip of the browser, also the `<a>` tag becomes no-different than a `div`

Comment (stanislav-slavov) [reply src/components/tipRecords/TipRecord.vue:38]: Because we prevent the click action from bubbling, otherwise the link gets its href executed and always open a new tab. In this case we want to open the dialog

File: src/components/TipControl.vue [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+          class="input-group"
+        >
+          <input
+            v-model="message"
+            type="text"
+            class="form-control tip__message"
+            placeholder="Add message"
+          >
+        </div>
+        <div v-if="!showLoading">
```
Comment (venimus) [inline src/components/TipControl.vue:53]: I would have made that a `<form>` with `@onsubmit="submitAction()` and put the `<ae-button>` to render a `button type=submit`. This way pressing Enter will work also will be somehow better supported on mobile

File: src/components/TipControl.vue [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      USE_DEEP_LINKS,
+      message: '',
+    };
+  },
+  computed: {
+    ...mapGetters(['account', 'balance', 'isLoggedIn', 'loading']),
+    eventPayload() {
+      return `${this.tip.id}:${this.show}`;
+    },
+    // deepLink() {
```
Comment (venimus) [inline src/components/TipControl.vue:110]: if it is not used, better not leave "commented probably-features"

Comment (mradkov) [reply src/components/TipControl.vue:110]: The deeplink should be used. Maybe @davidyuk can take a look at the flow and see if it needs adjustment.

File: src/components/tipRecords/TipRecord.vue (hunk 1 above)
Comment (venimus) [reply src/components/tipRecords/TipRecord.vue:38]: It should not happen if you use `@click.stop="handler"`

File: src/components/TipControl.vue (hunk 3 above)
Comment (venimus) [reply src/components/TipControl.vue:110]: so better tag it WIP

File: src/components/TipControl.vue [hunk 4]
```diff
@@ -154,14 +172,13 @@ export default {
     async sendTip() {
       const amount = util.aeToAtoms(this.value);
       this.showLoading = true;
-      await aeternity.tip(this.tip.url, this.title, amount)
+      await aeternity.tip(`${window.location.origin}/#/tip/${this.tip.id}`, this.message, amount)
```
Comment (venimus) [inline src/components/TipControl.vue:175]: can we first consider to use url without the domain. I'm very concerned about leaking random internal urls stored in the blockchain.

Comment (venimus) [reply src/components/TipControl.vue:175]: also it makes this untestable

Comment (mradkov) [reply src/components/TipControl.vue:175]: URL without the domain is not URL, rather a path.

File: src/components/AeButton.vue [hunk 5]
```diff
@@ -2,6 +2,7 @@
   <button
     :disabled="disabled"
     class="btn btn-primary ae-button"
+    type="submit"
     @click="$emit('click', $event)"
```
Comment (venimus) [inline src/components/AeButton.vue:6]: is this something functional?

File: src/components/TipControl.vue [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        <loading :show-loading="showLoading" />
+        <div
+          v-show="error && !showLoading"
+          class="text-center mb-2"
+        >
+          An error occurred while sending tip
+        </div>
+        <form
+          v-if="!showLoading"
+          @submit.prevent
```
Comment (venimus) [inline src/components/TipControl.vue:62]: doesn't this work `@submit.prevent="submitAction()"` (and remove it from the button)? Enter on the focused input emits submit event to the form, so the form should handle it. The same way button=submit emits the same event to the form, so we usually handle the event there. Then other onclick stuff become useless (including the one in ae-button). If `<ae-button>` is to be used as generic button, so we better make type to be configurable (as component property)

File: src/components/AeInputAmount.vue [hunk 7]
```diff
@@ -11,7 +11,7 @@
       aria-label="Default"
       aria-describedby="inputGroup-sizing-mn"
       @input="$emit('input', $event.target.value)"
```
Comment (venimus) [inline src/components/AeInputAmount.vue:13]: Not sure about these either

File: src/components/AeButton.vue [hunk 8]
```diff
@@ -2,6 +2,7 @@
   <button
     :disabled="disabled"
     class="btn btn-primary ae-button"
+    type="submit"
```
Comment (venimus) [inline src/components/AeButton.vue:5]: I think its best this to be passed as a component prop

File: src/components/Retip.vue [hunk 9]
```diff
@@ -170,33 +170,9 @@ export default {
...
-  .ae-amount {
-    align-items: center;
-    display: flex;
-    height: 1rem;
-  }
-
-  .retip__content {
+<style lang="scss">
+  .retip__content,
+  .tip__content {
```
Comment (venimus) [inline src/components/Retip.vue:175]: move all tip__content related css to the other component

File: src/components/TipControl.vue [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+          >
+            <input
+              v-model="message"
+              type="text"
+              class="form-control tip__message"
+              placeholder="Add message"
+            >
+          </div>
+          <div class="amount__row">
+            <ae-input-amount v-model="value" />
```
Comment (venimus) [inline src/components/TipControl.vue:75]:
```suggestion
            <ae-input-amount v-model.number="value" />
```

Comment (venimus) [reply src/components/TipControl.vue:75]: same should be to retip

Comment (venimus) [reply src/components/TipControl.vue:75]: Else, there is a validation issue: when you edit the number field and return it to 0, it becomes '0' (as string) and validation starts failing
